validator/src/main.rs

```diff
@@ -0,0 +1,261 @@
+use std::{
+    net::{IpAddr, SocketAddr},
+    process::exit,
+    sync::Arc,
+    time,
+};
+
+use anyhow::{anyhow, Context, Result};
+use clap::Parser;
+use rand::distributions::{Alphanumeric, DistString};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{TcpListener, TcpStream},
+    sync::Notify,
+};
+use tracing::{debug, error, info, Instrument};
+
+#[derive(Parser)]
+#[clap(version)]
+struct Args {
+    #[clap(
+        long,
+        env = "CNI_VALIDATOR_LOG_LEVEL",
+        default_value = "cni_validator=info,warn"
+    )]
+    log_level: kubert::LogFilter,
+
+    #[clap(long, default_value = "plain")]
```

Comment:
nit: other stuff also lets this be set from an env var, should we add
```suggestion
    #[clap(
        long,
        env = "CNI_VALIDATOR_LOG_FORMAT",
        default_value = "plain",
    )]
```